src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -195,3 +198,44 @@ SyntheticEvent.augmentClass = function(Class, Interface) {
 PooledClass.addPoolingTo(SyntheticEvent, PooledClass.fourArgumentPooler);
 
 module.exports = SyntheticEvent;
+
+
+// Utility functions
+
+/**
+  * Helper to nullify syntheticEvent instance properties when destructing
+  *
+  * @param {object} SyntheticEvent
+  * @param {String} propName
+  */
+function getPooledWarningPropertyDefinition(propName, getVal) {
+  var setTo = typeof getVal === 'function' ? 'a no-op function' : 'set to null';
+  return {
+    configurable: true,
+    set: function(val) {
+      var warningCondition = false;
+      warning(
+        warningCondition,
+        'This synthetic event is reused for performance reasons. If you\'re ' +
+        'seeing this, you\'re setting property `%s` on a ' +
+        'released/nullified synthetic event. This is effectively a no-op. See ' +
+        'https://fb.me/react-event-pooling for more information.',
+        propName
+      );
+      return val;
+    },
+    get: function() {
+      var warningCondition = false;
+      warning(
+        warningCondition,
+        'This synthetic event is reused for performance reasons. If you\'re ' +
+        'seeing this, you\'re accessing property `%s` on a ' +
+        'released/nullified synthetic event. This is %s. See ' +
+        'https://fb.me/react-event-pooling for more information.',
```

Comment:
> In addition, it might be best to preserve the old wording (calling a method rather than accessing the property) for methods.

I considered that, however this wouldn't make sense in a scenario where they're simply getting a reference to the method:

```javascript
const stop = event.stopPropogation
// maybe use stop later or something?
```

This would log a warning that wouldn't make sense because they're not actually calling it. I realize that's an edge case, but thought it would make the code simpler and the messaging more accurate.

Definitely willing to be overruled though :-)